homeassistant/components/media_player/cambridgeaudio.py

```diff
@@ -0,0 +1,366 @@
+
+    @property
+    def media_content_type(self):
+        """Return the media content type."""
+        return MEDIA_TYPE_MUSIC
+
+    @property
+    def media_image_url(self):
+        """Return image URI of current playing media."""
+        return self._album_art
+
+    @property
+    def media_artist(self):
+        """Return artist of current playing media."""
+        return self._artist
+
+    @property
+    def media_album_name(self):
+        """Return album of current playing media."""
+        return self._album
+
+    @property
+    def media_track(self):
+        """Return track number of current playing media."""
+        return self._trackno
+
+    @property
+    def media_title(self):
+        """Return title of current playing media."""
+        return self._title
+
+    @property
+    def media_position(self):
+        """Return position of current playing media in seconds."""
+        if self._state is not STATE_PLAYING:
+            return None
+        return self._position
+
+    @property
+    def media_duration(self):
+        """Return the duration of current playing media in seconds."""
+        if self._state is not STATE_PLAYING:
+            return None
+        return self._duration
+
+    @property
+    def media_position_updated_at(self):
+        """When was the position of the current playing media valid.
+
+        Returns value from homeassistant.util.dt.utcnow().
+        """
+        if self._state in [STATE_PLAYING, STATE_PAUSED]:
+            return self._position_updated_at
+
+    @property
+    def volume_level(self):
+        """Current volume level (0..1)."""
+        if self._support_volumecontrol:
+            return float(self._volume) / float(self._volume_max)
+
+    def clear_playlist(self):
+        """Clear the playlist."""
+        pass
+
+    def media_next_track(self):
+        """Skip to the next track, when in media player mode."""
+        if self._audio_source == "media player":
+            self._smdevice.trnsprt_next()
+
+    def media_previous_track(self):
+        """Skip to the previous track, when in media player mode."""
+        if self._audio_source == "media player":
+            self._smdevice.trnsprt_prev()
+
+    def media_pause(self):
+        """Pause playing the current media."""
+        # there is no "pause" with internet radio streams
+        if self._audio_source == "media player":
+            self._smdevice.trnsprt_pause()
+        elif self._audio_source == "internet radio":
+            self.media_stop()
+
+    def media_seek(self, position):
+        """Jump to the specified percent position within the current track."""
+        if self._audio_source != "media player":
+            return
+
+        abs_pos = round(self._to_seconds(self._duration) * (position/100), 2)
+        abs_pos_h = int(abs_pos / 3600)
+        abs_pos_m = int((abs_pos % 3600) / 60)
+        abs_pos_s = int((abs_pos % 3600) % 60)
+        abs_pos = '{:01d}:{:02d}:{:02d}'.format(abs_pos_h,
```

Comment:
I think you can convert to a time object first then do the time manipulation, then convert back to time formatted string. First try to use the `parse_time` util in util/dt.py, or something else in there.

Next option is to use available class methods in datetime module.

https://docs.python.org/3.6/library/datetime.html#strftime-strptime-behavior

Reply:
For the _to_seconds() method, I could use dt_util.parse_time() to get a datetime.time object.
As far as I can see, any strftime() format would only give me seconds between 0..59, though.
So I would need to create a datetime.timedelta object from it and call its .total_seconds() method, e.g.:
```
import datetime as dt
t = dt.time(1, 10, 10)
int(dt.timedelta(hours=t.hour, minutes=t.minute, seconds=t.second).total_seconds())
4210
```
That feels overly complicated compared to my one-line helper function.

I'll rewrite the media_seek() method to using time.strftime().